tests/commands/test_build.py

```diff
@@ -319,7 +319,11 @@ def test_politeexec_stdout_logged(caplog):
 
     cmd = ['echo', 'HELO']
     polite_exec(cmd)
-    assert [":: HELO"] == [rec.message for rec in caplog.records]
+    expected = [
+        "Running external command ['echo', 'HELO']",
```

Comment:
This isn't the nicest formatting for a user facing output, but if it is debug logging, then I suppose it is more precise.

Reply:
I was thinking about this, and I think we can print a more user-friendly error without losing the debugging ability by joining the commandline with some way that highlights the difference between spaces between arguments and spaces within arguments. It doesn't have to be perfect as it's just for logging. It could be something as simple as joining the command list with middle dots, so that `['echo', 'HELO WORLD']` becomes `'echo·HELO WORLD'`, or, given that space within args is rarer, it could be just `'echo HELO␠WORLD'`. Both of these are subtle so that they're easy to read if you don't need to know, and clear if you do. IMHO.

Reply:
FWIW that's `"·".join(['echo', 'HELO WORLD'])` vs `" ".join("␠".join(i.split()) for i in ['echo', 'HELO WORLD'])`

Reply:
(to be clear: this is something to be done later, across all places where we log commands)